### ISSUE TITLE
sensor.ariston_active_errors is displaying 0 

### ISSUE THREAD

**Issue by viomatei** [chomupashchuk/ariston-remotethermo-home-assistant-v2#43]
Hi, 

Just so happens that the heating water pressure dropped below 1 bar and I got IP4 error message on my Ariston Genus One display, but the sensor.ariston_active_errors, in HA, is displaying 0.
Any ideas?

**Comment by chomupashchuk** [comment]
Not really.
I believe that errors are fetched approximately every 2 minutes and shall not detect an error if it dissapears meanwhile. 
I triggered error once on my boiler and have written parsing of data based on it. It is not likely, but errors format could differ causing issues in parsing. Another potential possibility is that Ariston server clears error on server side after reading from any source. And another possibility that such errors are not reported in that specific request.
Did you check the logs for warnings or errors?

**Comment by viomatei** [comment]
There is no warnings or errors in the logs, and the IP4 error is still on the boiler's display (as I did not refill the pluming with water). 
I remember that the same error was being pulled in the 1.x versions (it's the reason I made an automation for when the pressure would drop, so I can get back to nominal state).

**Comment by chomupashchuk** [comment]
Add line after `self._error_data = copy.deepcopy(resp.json())`:
`self._LOGGER.warn(f'ERROR REPLY: {self._error_data}')`

When error occurs again then logs have to be checked. It is to see if request returns a valid reply with errors data within it and if it does then how data looks like.

If the request won't have an error inside then a new method of error collection would have to be investigated.

**Comment by chomupashchuk** [comment]
![errors](https://user-images.githubusercontent.com/50545689/180613713-8b84fc92-ac60-44fe-b484-4aba8cb5664c.png)

Is what I get in reply when I turned off Gas supply (interestingly Ariston started reporting it after ~1 minute when I checked it manually constantly and integration did require additional time due to time required to send all requests so took about 2 minutes total in my case).
`[{'gw': 'F0AD4E0590BD', 'timestamp': '2022-07-23T16:14:12', 'fault': 45, 'mult': 0, 'code': '501', 'pri': 1053500, 'errDex': 'No flame detected', 'res': False, 'blk': True}]`

**Comment by viomatei** [comment]
OK, it works. I turned off the gas supply and I got :Error_1: 2022-07-24T06:38:50, No flame detected
Don't know why the loss of pressure did not got trough, as this kind of error is important for my setup, but maybe as you said, could be a error on the server side. I don't have any option for release the pressure and check if the warning works, but I'll keep an eye on it.

Thank you again for your hard work and dedication. I really appreciated.
